src/openms/include/OpenMS/ANALYSIS/ID/AhoCorasickDA.h

```diff
@@ -0,0 +1,245 @@
+    const char* protein_ = "";
+
+    // position in protein
+    UInt prot_pos_ = 0;
+
+    // position in DA
+    Int32 node_pos_= 0;
+
+    std::vector<Spawn> spawns_{};
+
+
+    // code table
+    UInt code_tabel_ [256] =
+        { // char --> amino acid (unsigned char with values from 0..26)
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25, //0
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25, //1
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  26,  25,  25,  25,  25,  25, //2
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25, //3
+
+        //    ,   A,   B,   C,   D,   E,   F,   G,   H,   I,   J,   K,   L,   M,   N,   O,
+            25,   1,  22,   18,  8,   4,  14,   2,  17,   6,  24,   7,   0,  16,  12,  20, //4
+
+        //   P,   Q,   R,   S,   T,   U,   V,   W,   X,   Y,   Z,    ,    ,    ,    ,    ,
+            11,  13,   9,   5,  10,  21,   3,  19,  25,  15,  23,  25,  25,  25,  25,  25, //5
+
+        //    ,   a,   b,   c,   d,   e,   f,   g,   h,   i,   j,   k,   l,   m,   n,   o,
+            25,   1,  22,  18,   8,   4,  14,   2,  17,   6,  24,   7,   0,  16,  12,  20, //6
+
+        //   p,   q,   r,   s,   t,   u,   v,   w,   x,   y,   z,    ,    ,    ,    ,    ,
+            11,  13,   9,   5,  10,  21,   3,  19,  25,  15,  23,  25,  25,  25,  25,      //7
+
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25, //8
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25, //9
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25, //10
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25, //11
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25, //12
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25, //13
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25, //14
+            25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25,  25  //15
+        };
+
+
+
+    // returns code of label
+    UInt32 code_(const char& label);
+
+    // returns the index of child of the node with base_value with the transition label uses XOR
+    UInt32 child_(const UInt32& base_value, const char& label);
+
+    // calls functions to construct the Aho Corasick
+    void construct_();
+
+    // recursive function to create the trie by subtries
+    void buildTrie_(Size begin, const Size& end, const Size& depth, const UInt32& node_pos);
+
+
+    bool retrieval_();
+
+    //expands the BC array and initializes linkage
+    void expand_();
+
+
+    // returns base value
+    UInt32 xCheck_();
+    UInt32 plusCheck_();
+
+
+    // tests the base value for all labels in edges
+    bool checkAllEdges_(const UInt32& base);
```

Comment:
dont take a small type by ref.
Use `const UInt32`.

There are more functions which do this. please change them too.